plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -169,6 +191,59 @@ def get_models(self, metadata, mask_id=''):
         return images
 
 
+# ****************************************************************************************************
+class AddManifestList(PluginStep):
+    """
+    Add an uploaded ManifestList to a repository.
+    """
+
+    def process_main(self):
+        """
+        TODO(asmacdo) write
+        """
+        with open(self.parent.file_path, 'r') as uploaded_file:
+            manifest_list = uploaded_file.read()
+
+        expected_digest = self.parent.unit_key['digest']
+        digest = models.Manifest.calculate_digest(manifest_list, expected_digest.split(':')[0])
+        if digest != expected_digest:
```

Comment:
i understand that in docs we mention that unit_key is a required param, but as you pointed out earlier, it is unused, and because we are usually able extract the information on our own from the provided file, i don't see the point to enforce user to calculate the digest.
in addition to what was said above, during image manifest upload we don't enforce user to calculate the digest, we do it on his behalf, so if we would introduce this requirement in the upload of the manifest list i think it would be a surprise.

Reply:
Ok, makes sense. I'll update so we only validate the digest if the user provides one